extensions/Worker.Extensions.Http.AspNetCore/src/FunctionContextExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Provides extension methods to work with <see cref="FunctionContext"/>.
+    /// </summary>
+    public static class FunctionContextExtensions
+    {
+        /// <summary>
+        /// Retrieve the <see cref="HttpContext"/> for the function execution.
+        /// </summary>
+        /// <param name="context">The <see cref="FunctionContext"/>.</param>
+        /// <returns></returns>
+        public static HttpContext GetHttpContext(this FunctionContext context)
+        {
+            if (context.Items.TryGetValue(Constants.HttpContextKey, out var requestContext)
+                && requestContext is HttpContext httpContext)
+            {
+                return httpContext;
```

Comment:
We also have another `GetHttpRequestDataAsync` extension method which returns an `HttpRequestData` instance. Wonder we need to keep that in sync if users are updating http props in this HttpContext. Ex: If user adds another request header to the HttpContext.Request, should that be reflected in the HttpRequestData instance? Or customers won't be using the HttpRequestData type in this new world?

Reply:
I don't believe the two models should be combined - so `HttpRequestData` shouldn't be used with `HttpRequest`.

Reply:
The methods exist, so we'd either need to make it very clear it's not a supported scenario, or make sure those are in sync (i.e., return an `HttpRequestData` implementation backed by the `HttpContext`). Doesn't need to be part of this PR, but something we should track and flag in the initial preview.